src/arithmetic/algebraic_expression/algebraic_expression_optimization.c

```diff
@@ -24,6 +25,7 @@ static void _AlgebraicExpression_CollectOperands(AlgebraicExpression *root,
 	switch(root->type) {
 	case AL_OPERAND:
 		*operands = array_append(*operands, AlgebraicExpression_Clone(root));
+		root->operand.bfree = false; // The caller is the new owner of this operand.
```

Comment:
Can you please explain to me why are we resetting `bfree` ?
Who's calling `_AlgebraicExpression_CollectOperands` and for what purpose?

Reply:
`_AlgebraicExpression_FlattenMultiplications` is the sole caller of this function; it uses this function to collect n-ary operands under the same multiplication operation rather than having nested multiplication operations.

It frees the operations that have been made redundant, and as such could cause invalid frees of operands that have been migrated.

This is not a possible scenario currently, but logically it makes sense to reset `bfree` here.